evalml/model_understanding/feature_explanations.py

```diff
@@ -0,0 +1,208 @@
+        min_importance_threshold,
+        linear_importance,
+    )
+    target = y if y is None else y.name
+    explanation = _fill_template(
+        pipeline.estimator,
+        target,
+        objective,
+        most_important_features,
+        somewhat_important_features,
+        detrimental_features,
+    )
+    logger.info(explanation)
+
+
+def get_influential_features(
+    imp_df, max_features=5, min_importance_threshold=0.05, linear_importance=False
+):
+    """Finds the most influential and detrimental features from a dataframe of feature importances.
+
+    Args:
+        imp_df (pd.DataFrame): DataFrame containing feature names and associated importances.
+        max_features (int): The maximum number of features to include in an explanation. Defaults to 5.
+        min_importance_threshold (float): The minimum percent of total importance a single feature can have to be considered important. Defaults to 0.05.
+        linear_importance (bool): When True, negative feature importances are not considered detrimental. Defaults to False.
+
+    Returns:
+        (list, list, list): Lists of feature names corresponding to heavily influential, somewhat influential, and detrimental features, respectively.
+
+    """
+    heavy_importance_threshold = max(0.2, min_importance_threshold + 0.1)
+
+    # Separate negative and positive features, if situation calls
+    if linear_importance:
+        pos_imp_df = imp_df
+        pos_imp_df["importance"] = abs(pos_imp_df["importance"])
+        neg_imp_df = pos_imp_df[pos_imp_df["importance"] < 0]
+    else:
+        neg_imp_df = imp_df[imp_df["importance"] < 0]
+        pos_imp_df = imp_df[imp_df["importance"] >= 0]
+
+    # Normalize the positive features to sum to 1
+    pos_imp_df["importance"] = pos_imp_df["importance"] / sum(pos_imp_df["importance"])
+
+    num_feats = min(len(pos_imp_df), max_features)
+    imp_features = pos_imp_df[:num_feats]
+    return (
+        list(
+            imp_features[imp_features["importance"] >= heavy_importance_threshold][
+                "feature"
+            ]
+        ),
+        list(
+            imp_features[
+                imp_features["importance"].between(
+                    min_importance_threshold,
+                    heavy_importance_threshold,
+                    inclusive="left",
+                )
+            ]["feature"]
+        ),
+        list(neg_imp_df["feature"]),
+    )
+
+
+def _fill_template(
+    estimator, target, objective, most_important, somewhat_important, detrimental_feats
+):
+    if objective is not None:
+        if target is not None:
+            beginning = f"{estimator}: The performance of predicting {target} as measured by {objective}"
+        else:
+            beginning = f"{estimator}: The {objective} performance"
+    else:
+        if target is not None:
+            beginning = f"{estimator}: The prediction of {target}"
+        else:
+            beginning = f"{estimator}: The output"
```

Comment:
I wonder if this can be cleaned up just a tiny bit:

```
objective_str = "as measured by {objective}" if objective is not None else ""
beginning_str = "The output {objective_str}" if target is None else "The prediction of target {objective_str}"
```     


if objective is None, target is None: beginning_str = The output
if objective is None, target is not None: beginning_str = The prediction of target 
if objective is not None, target is None: beginning_str = The output as measured by objective
if objective is not None, target is not None: The prediction of target as measured by objective...

Something like that?